src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java

```diff
@@ -924,7 +923,10 @@ protected void enqueueAndHandleVariant(final VariantContext variant, final Refer
 
         final FuncotationMap funcotationMap = funcotatorEngine.createFuncotationMapForVariant(variant, referenceContext, featureContext);
 
+        // This is necessary because we want to revert the variant contig namechange if it was applied in the VariantWalkerBase::apply method before output.
```

Comment:
Well, we don't _always_ want to revert the contig name change, right? You should clarify in the comment here under which circumstances we need to revert.

Also, the contig name change happens in the variant transformer, not `VariantWalkerBase::apply`